src/jvm/main/org/jetbrains/kotlinx/lincheck/transformation/LincheckClassVisitor.kt

```diff
@@ -0,0 +1,1407 @@
+/*
+ * Lincheck
+ *
+ * Copyright (C) 2019 - 2024 JetBrains s.r.o.
+ *
+ * This Source Code Form is subject to the terms of the
+ * Mozilla Public License, v. 2.0. If a copy of the MPL was not distributed
+ * with this file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package org.jetbrains.kotlinx.lincheck.transformation
+
+import org.jetbrains.kotlinx.lincheck.TransformationClassLoader.ASM_API
+import org.jetbrains.kotlinx.lincheck.Injections
+import org.jetbrains.kotlinx.lincheck.strategy.managed.JavaUtilRemapper
+import org.objectweb.asm.*
+import org.objectweb.asm.Opcodes.*
+import org.objectweb.asm.Type.*
+import org.objectweb.asm.commons.*
+import org.objectweb.asm.commons.InstructionAdapter.*
+import org.jetbrains.kotlinx.lincheck.transformation.CoroutineInternalCallTracker.isCoroutineInternalClass
+import java.util.*
+
+internal class LincheckClassVisitor(
+    private val transformationMode: TransformationMode,
+    cw: ClassVisitor
```

Comment:
What is `cw`?

Reply:
classVisitor, renamed

Reply:
What is this classVisitor? Is it always a `ClassWriter` instance?